### PR TITLE
Package doc (doc.go)

### PR DESCRIPTION
**Pull request by mattdeekay**
Documentation in Go package docs so that they will appear in https://pkg.go.dev/github.com/databricks/databricks-sql-go. Sections include:
- Usage
- Connection via DSN string
- Connection via connector object
- Query cancellation, timeout
- CorrelationId and ConnId
- Logging

### CODE REVIEW

**Comment by andrefurlan-db** [comment]
Add some info on data type support as well? Specially on structs, maps, and arrays; decimal.

**Comment by yunbodeng-db** [comment]
@andrefurlan-db can you review this change? Thanks!

**Review by andrefurlan-db — CHANGES_REQUESTED** [review]

File: doc.go [hunk 1]
```diff
@@ -1,4 +1,198 @@
...
+
+# Supported Data Types
+
+==================================
+
+Thrift Server Type --> Golang Type
+
+==================================
+
+TTypeId_BOOLEAN_TYPE      --> bool
```
[the remark this answers is not in the document]
Comment (mattdeekay) [reply doc.go:154]: Fair ~ will change

File: doc.go [hunk 2]
```diff
@@ -1,4 +1,190 @@
...
+
+  - catalog: Sets the initial catalog name in the session
+  - schema: Sets the initial schema name in the session
+  - maxRows: Sets up the max rows fetched per request. Default is 10000
+  - timeout: Adds timeout (in seconds) for the server query execution. Default is no timeout
+  - userAgentEntry: Used to identify partners. Set as a string with format <isv-name+product-name>
+
+Supported optional session parameters can be specified in param=value and include:
+
+  - ansi_mode: (Boolean string). Session statements will adhere to rules defined by ANSI SQL specification. Default is "false"
```
Comment (andrefurlan-db) [inline doc.go:43]: oh sorry, I think the default for data warehouses is `true`

File: doc.go [hunk 3]
```diff
@@ -1,4 +1,190 @@
...
+
+DATE --> time.Time
+
+TIMESTAMP --> time.Time
+
+DECIMAL(p,s) --> sql.RawBytes
+
+BINARY --> sql.RawBytes
+
+ARRAY<elementType> --> sql.RawBytes
```
Comment (andrefurlan-db) [inline doc.go:180]: maybe worth mentioning that ARRAY, STRUCT, MAP can be scanned to a string, in which will be like:
`{arrayVal:[1,2,3] mapVal:{"key1":1} structVal:{"string_field":"string_val","array_field":[4,5,6]}}`
And from there the user can unmarshal the JSON string to the appropriate type

**Review by andrefurlan-db — APPROVED** [review]

**Pull request merged**